lib/sha256/CMakeLists.txt

```
@@ -0,0 +1,4 @@
+add_library(sha256 STATIC sha256.c)
+
+set(SHA256_INCLUDE_DIR ${CMAKE_CURRENT_SOURCE_DIR} PARENT_SCOPE)
+set(SHA256_LIBRARY sha256 PARENT_SCOPE)
```

```suggestion
target_include_directories(sha256 INTERFACE .)
```
CMake will automatically pass this to any target that will link to `sha256`.

And if you want to be really absolutely sure you link to a target and not just a library, you may add an alias target like `add_library(sha::sha256 ALIAS sha256)` and link MT to `sha::sha256` instead.